### ISSUE TITLE
azurerm_storage_share_file content_md5 incompatible with filemd5() - re-open

### ISSUE THREAD

**Issue by oskarm93** [hashicorp/terraform-provider-azurerm#28058]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.5

### AzureRM Provider Version

3.117.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_share_file

### Terraform Configuration Files

```hcl
import {
  for_each = fileset(local.logstash_files_dir, "*.conf")
  id       = "${azurerm_storage_share.logstash_files.id}/${each.key}"
  to       = azurerm_storage_share_file.logstash_config_files[each.key]
}

resource "azurerm_storage_share_file" "logstash_config_files" {
  for_each         = fileset(local.logstash_files_dir, "*.conf")
  name             = each.key
  storage_share_id = azurerm_storage_share.logstash_files.id
  source           = "${local.logstash_files_dir}/${each.key}"
  content_md5      = filemd5("${local.logstash_files_dir}/${each.key}")
}
```
```


### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

For this to be fixed since v3.115.0. This issue has been opened, fixed and closed. I cannot re-open it. And yet, in v3.117.0 in the provider I am still encountering the problem.

https://github.com/hashicorp/terraform-provider-azurerm/issues/25704

### Actual Behaviour

`content_md5` property still shows hex string instead of base64 after importing the resource, and replacing it twice (due to md5 hash not matching). 
Hashes are matching, just in the wrong representation: hex string vs base64.

PowerShell:
`[convert]::ToHexString([convert]::FromBase64String('qcTcld85Ja4zNyc1VcHTbw==')).ToLower()`

### Steps to Reproduce

```
terraform init

Initializing provider plugins...
- Finding hashicorp/azurerm versions matching "< 4.0.0"...
- Installing hashicorp/azurerm v3.117.0..
- Installed hashicorp/azurerm v3.117.0 (signed by HashiCorp)
```

```
terraform plan # first time

# azurerm_storage_share_file.logstash_config_files["empty.conf"] must be replaced
  # (imported from "[xxx/logstash-files/empty.conf"](xxx/logstash-files/empty.conf%22))
  # Warning: this will destroy the imported resource
-/+ resource "azurerm_storage_share_file" "logstash_config_files" {
      ~ content_length   = 154 -> (known after apply)
        content_md5      = "a9c4dc95df3925ae3337273555c1d36f"
        content_type     = "application/octet-stream"
      ~ id               = "[xxx/logstash-files/empty.conf"](xxx/logstash-files/empty.conf%22) -> (known after apply)
      - metadata         = {} -> null
        name             = "empty.conf"
      + source           = "./logstash-config-files/empty.conf" # forces replacement
        storage_share_id = "[xxx/logstash-files"](xxx/logstash-files%22)
    }
```

```
terraform plan # second time

  # azurerm_storage_share_file.logstash_config_files["empty.conf"] must be replaced
-/+ resource "azurerm_storage_share_file" "logstash_config_files" {
      ~ content_length   = 154 -> (known after apply)
      ~ content_md5      = "qcTcld85Ja4zNyc1VcHTbw==" -> "a9c4dc95df3925ae3337273555c1d36f" # forces replacement
      ~ id               = "xxx/logstash-files/empty.conf" -> (known after apply)
      - metadata         = {} -> null
        name             = "empty.conf"
        # (3 unchanged attributes hidden)
    }
```

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/25704
#25704

**Comment by oskarm93** [comment]
@magodo I think you had the original PR to fix the issue. `content_md5` is set correctly as base64 in the state file but it looks like the input is still a hex string.

Workaround for now:
```
resource "terraform_data" "logstash_config_files" {
  for_each = fileset(local.logstash_files_dir, "*.conf")
  input    = filemd5("${local.logstash_files_dir}/${each.key}")
}

resource "azurerm_storage_share_file" "logstash_config_files" {
  for_each         = fileset(local.logstash_files_dir, "*.conf")
  name             = each.key
  storage_share_id = azurerm_storage_share.logstash_files.id
  source           = "${local.logstash_files_dir}/${each.key}"
  lifecycle {
    replace_triggered_by = [terraform_data.logstash_config_files[each.key]]
  }
}
```

https://developer.hashicorp.com/terraform/language/resources/terraform-data#example-usage-data-for-replace_triggered_by

**Comment by magodo** [comment]
Hi @oskarm93, sorry for running into this. There is a follow up PR to fix this issue https://github.com/hashicorp/terraform-provider-azurerm/pull/27108. Unfortunately, this fix is included in v4.0.0, and isn't back ported to v3.

Please let me know if you are fine to use the v4 provider, otherwise I'll bring this to the team to see how we can help.
